src/devices/thermostat.cpp

```diff
@@ -679,12 +671,21 @@ void Thermostat::process_RC10Monitor(std::shared_ptr<const Telegram> telegram) {
 
 // add the HVAC/Climate HA component for the HC
 void Thermostat::add_ha_climate(std::shared_ptr<HeatingCircuit> hc) {
-    if (!Mqtt::ha_enabled() || (hc->ha_climate_created())) {
+    if (!Mqtt::ha_enabled()) {
         return;
     }
 
-    // only if it has a valid seltemp (roomtemp is optional)
-    if (Helpers::hasValue(hc->selTemp)) {
+    bool has_values = Helpers::hasValue(hc->selTemp) && Helpers::hasValue(hc->roomTemp);
+
+    // see if we need to remove the HA climate component because some of the values have been lost
+    // note, this doesn't account for whether any of the device values have been excluded
+    if (hc->ha_climate_created() && !has_values) {
+        // remove the whole climate component
+        char topic[Mqtt::MQTT_TOPIC_MAX_SIZE];
+        snprintf(topic, sizeof(topic), "climate/%s/thermostat_hc%d/config", Mqtt::base().c_str(), hc->hc_num());
+        Mqtt::publish_ha(topic);
+    } else {
+        // create
         publish_ha_config_hc(hc);
         hc->ha_climate_created(true); // only create it once
     }
```

Comment:
Is it intended that this publishes/removes in every call, every 10 sec on normal monitor-broadcast?

Reply:
I'll fix it

Reply:
Hmm, now it is created once, removed if seltemp is missing, but never created again. What about roomttemp now?
I think better is to remove ond re-add:
```
    if (hc->ha_climate_created()) {
        if (!has_values) {
            // remove the whole climate component
            char topic[Mqtt::MQTT_TOPIC_MAX_SIZE];
            snprintf(topic, sizeof(topic), "climate/%s/thermostat_hc%d/config", Mqtt::base().c_str(), hc->hc_num());
            Mqtt::publish_ha(topic);
            hc->ha_climate_created(false); // remove flag, can be added again
        }
    } else if (has_values) {
        // create
        publish_ha_config_hc(hc);
        hc->ha_climate_created(true);
    }

```
Is it a solution for missing roomtemp (outdoor controlled) and HA enabled to set `hc->roomTemp = hc->selTemp;`

Reply:
if seltemp goes, it will be created and it should be created again with the check ha_climated_created()

the roomtemp is optional and handled in the publish_ha_config() function

Reply:
The config is created regardless of seltemp and afterwards removed if seltemp is not set. If seltemp is coming back, it is NOT recreated, because ha_climate_created() stays true, it is only set to false on hc creation. 

But seltemp is the major value of thermostat, it will never be absent, without seltemp a thermostat is no thermostat. So this code will never be executed and works as before. 

I think you can merge this PR to add the other changes. We come back later to the climate creation in #371, which need some more adaptations.